api/all/src/main/java/io/opentelemetry/api/baggage/ImmutableEntryMetadata.java

```diff
@@ -24,6 +25,7 @@
    */
   static ImmutableEntryMetadata create(String metadata) {
     if (metadata == null) {
+      ValidationUtil.log("metadata is null");
```

Comment:
I wonder if we should provide some sort of contextual information with these log entries. If I saw this in my logs I might not know what would be the source of the logging, since all of them will be under the same logger name. I guess the AssertionError that gets logged would have the stack trace in it?

Reply:
Yeah think the stack trace should be enough context unless there's anything else that comes to mind to add. But don't think we need to repeat method information.

Reply:
I have just verified to confirm the stacktrace is logged